Cutelyst/action.cpp

```diff
@@ -207,6 +210,139 @@ bool Action::doExecute(Context *c)
+                break;
+            case 1:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c);
+                break;
+            case 2:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0));
+                break;
+            case 3:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1));
+                break;
+            case 4:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2));
+                break;
+            case 5:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3));
+                break;
+            case 6:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3), args.value(4));
+                break;
+            case 7:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3), args.value(4), args.value(5));
+                break;
+            case 8:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3), args.value(4), args.value(5), args.value(6));
+                break;
+            case 9:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3), args.value(4), args.value(5), args.value(6), args.value(7));
+                break;
+            default:
+                ret = d->method.invoke(d->controller, Qt::DirectConnection, qReturnArg(methodRet), c, args.value(0), args.value(1), args.value(2), args.value(3), args.value(4), args.value(5), args.value(6), args.value(7), args.value(8));
+                break;
+            }
+        }
+
+        if (ret) {
+            c->setState(methodRet);
+            return methodRet;
+        }
+
+        // The method failed to be called which means we should detach
+        c->detach();
+        c->setState(false);
+
+        return false;
+    } else {
+        if (d->listSignature) {
+
+            QStringList args = c->request()->args();
+
+            ret = d->method.invoke(d->controller,
+                                   Qt::DirectConnection,
+                                   c,
+                                   args);
+        } else {
+            QStringList args = c->request()->args();
+            // Fill the missing arguments
+            args.append(d->emptyArgs);
```

Comment:
and this

Reply:
Ok

Reply:
@Danxomme No, not you. :) This was meant for me.

Reply:
OK